datatypes.go

```diff
@@ -115,6 +133,36 @@ type TriggerData struct {
 	Tags       []string `json:"__notifier_trigger_tags"`
 }
 
+func (trigger TriggerData) PopulateDescription(events NotificationEvents) (string, error) {
+	buffer := new(bytes.Buffer)
+	templateEvents := make([]templateEvent, 0, len(events))
+
+	for _, data := range events {
+		event := templateEvent{
+			Metric:         data.Metric,
+			MetricElements: strings.Split(data.Metric, "."),
+			Timestamp:      data.Timestamp,
+			State:          data.State,
+		}
+
+		if data.Value != nil {
```

Comment:
Why do you need to check for `nil` here? What's going to change if you remove this check?